text/0036-client-network-variables.md

```diff
@@ -0,0 +1,142 @@
+- The server then broadcasts this to the other clients
+- Note, unlike pre-1.0 versions of NGO when clients could write to network variables, the server can **not** write to client network variables.
+
+## Private Mode
+
+So that developers can choose to replicate the `ClientNetworkVariable` just on the server but **not** to the other clients, a `OwnerOnly` mode is avaiable.
+
+For example, one might want to use the `ClientNetworkVariable` to transmit a "cast spell" input command to just the server (vs. today where an RPC is required) so that it can react to the input and then update the other clients accordingly (e.g., update the clients differently whether the user actually has manna or not).  To do this, one declares the `ClientNetworkVariable` with the `OwnerOnly` permission, e.g.
+
+
+```
+public class NetworkVariableTest : NetworkBehaviour
+{
+    public ClientNetworkVariable<int> SomePrivateClientVar = 
+        new ClientNetworkVariable<int>(NetworkVariableReadPermission.OwnerOnly);
+
+    // ...
+
+    void Update()
+    {
+        if (NetworkManager.isClient)
+        {
+            // only this server and the sending client will know about this
+            SomePrivateClientVar.Value = 456; 
+        }
+
+    }
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why might we *not* do this?
+See the "Future possibilities" for a more detailed explanation, but we likely will need to change how this system works when moving on to a snapshotted model in future major releases.  In the current system there is no canonical "on tick 123 these variables had values {...}", and so having the client broadcast values at some arbitrary time to be received by clients at some aribitrary time is not disriptive.  As a result users used to this model may have to adapt to something different in the next major release.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This feature is a building block one can use for client-authoritative features.  However it intentionally leaves the "local manifestation" element to the user.  That is, while a user can define the state and the SDK will sync it, it is up to the user whether to take that `ClientNetworkVariables` state and manifest it, say, in a character's position OR transmit it as a command to the server and wait for the server's updates.
+
+In considering the feature and talking with developers the common feedback was this was the level of involvement they wanted from the SDK.  Moreover, while we felt reasonably confident that we could define variable use case, we were less certain we could define a full client authority system with local manifestation given all the different use cases.  Note however such additions could be added later, building on this system. 
+
+Technial advantages of this approach:
+- programmer interface is just like before
```

Comment:
Before, we had the same netvar for both server and client, but with permissions. People porting to Netcode 1.x from 0.1.0 would still have to create new code to use the new ClientNetworkVariable.